src/pages/api/conversation.ts

```diff
@@ -283,179 +45,71 @@ async function handler(req: AuthenticatedRequest, res: NextApiResponse) {
 
         // Convert conversation to DB type
         const dbConversation = convertChatToDBConversation(conversation)
-
-        if (conversation.messages.length === 0) {
-          // Return success without saving - no need to throw an error
-          return res.status(200).json({ message: 'No messages to save' })
-        }
-
-        // Create a correctly typed conversation object for DrizzleORM
-        const conversationData: NewConversations = {
-          id: dbConversation.id,
-          name: dbConversation.name,
-          model: dbConversation.model,
-          prompt: dbConversation.prompt,
-          temperature: dbConversation.temperature,
-          user_email: user_identifier || null,
-          project_name: dbConversation.project_name,
-          folder_id: isUUID(dbConversation.folder_id ?? '')
-            ? dbConversation.folder_id
-            : null,
-          created_at: dbConversation.created_at
-            ? new Date(dbConversation.created_at)
-            : new Date(),
-          updated_at: dbConversation.updated_at
-            ? new Date(dbConversation.updated_at)
-            : new Date(),
-        }
-
-        // Save conversation to DB using DrizzleORM
-        try {
+        await upsertConversationRecord(dbConversation, user_identifier)
+        // Save messages using messages API to maintain backward compatibility
+        for (const message of conversation.messages) {
```

Comment:
This part still assumes it gets the message in the request body, but the change that was made in the call from client, strips that off. I havent tested the PR but I found this as I was reading the diffs.